### ISSUE TITLE
GAP | NSFS | Versioning | List Object Version - Always Returns `IsLatest` true

### ISSUE THREAD

**Issue by shirady** [noobaa/noobaa-core#7372]
### Environment info
- NooBaa Version: current master (5.14)
- Platform: Rancher Desktop 1.19.0 (VM and cluster)

### Actual behavior
1. We always return `IsLatest` true.

### Expected behavior
1. Only the latest version should return `IsLatest` true.

#### General Settings:
1. Change the configuration of NSFS Versioning to be Enabled in file `config.js` and change `config.NSFS_VERSIONING_ENABLED` to `true`.
2. Deploy noobaa system on minikube (Based on the instructions [here](https://gist.github.com/dannyzaken/c85f0006e5d6582c8d8666cafce6ab76) - the steps of _‘Build images’_ and _‘Deploy noobaa’_).
_Note: `nb` is an alias that runs the local operator from `build/_output/bin` (alias created by `devenv`)._
4. Deploy NSFS on minikube (Based on the instructions [here](https://github.com/noobaa/noobaa-core/wiki/NSFS-on-Kubernetes/8fa95c3b827c40f2435a63f60fae5f7285e6b221)).
#### Steps:
5. Enabling versioning
`aws --endpoint-url=https://localhost:12443 --no-verify-ssl s3api put-bucket-versioning --bucket fs1-jenia-bucket --versioning-configuration Status=Enabled`
6. Put an object twice
` s3-nb-user-1 aws --endpoint-url=https://localhost:12443 --no-verify-ssl s3api put-object --bucket fs1-jenia-bucket --key fish.txt --body <some_txt_file>`
` s3-nb-user-1 aws --endpoint-url=https://localhost:12443 --no-verify-ssl s3api put-object --bucket fs1-jenia-bucket --key fish.txt --body <some_txt_file>`
7. List object versions:
`aws --endpoint-url=https://localhost:12443 --no-verify-ssl s3api list-object-versions --bucket fs1-jenia-bucket`

### More information - Screenshots / Logs / Other output
Noobaa output:
![Screenshot 2023-07-02 at 16 58 36](https://github.com/noobaa/noobaa-core/assets/57721533/70d12453-5efb-4647-91a3-0c933a39301c)

AWS output:
![Screenshot 2023-07-02 at 17 09 36](https://github.com/noobaa/noobaa-core/assets/57721533/0940b89b-4727-49ce-a183-c555ed64d82e)

**Comment by shirady** [comment]
@v-harihar knows about this gap, so I edited the title.

**Comment by hseipp** [comment]
Please note that in addition to the object versions also multiple delete markers for a single object show IsLatest as true .
After deleting a single object, re-adding a new version and again deleting it:

```
...
    "DeleteMarkers": [
        {
            "Owner": {
                "DisplayName": "NooBaa",
                "ID": "123"
            },
            "Key": "key1",
            "VersionId": "mtime-d3lql1gt4ruo-ino-t1g",
            "IsLatest": true,
            "LastModified": "2024-08-21T16:39:42+00:00"
        },
        {
            "Owner": {
                "DisplayName": "NooBaa",
                "ID": "123"
            },
            "Key": "key1",
            "VersionId": "mtime-d3lqj85nwwlc-ino-t1e",
            "IsLatest": true,
            "LastModified": "2024-08-21T16:37:19+00:00"
        }
    ],
...
```

**Comment by romayalon** [comment]
Closing this issue per Nadav's fix - @nadavMiz @8298

**Comment by rkomandu** [comment]
is this backported into 4.17, it is not available in the 4.17-0904 d/s build

**Comment by romayalon** [comment]
@rkomandu 
This is the link to check if a fix was backported to 5.17 - https://github.com/noobaa/noobaa-core/commits/5.17/
Search the PR's commit name - [NSFS | NC | fix list-objecs-versions issues](https://github.com/noobaa/noobaa-core/pull/8298/commits/b21517a5256d089f6fc7a308fec3688ce9d740c9)
and then you can see it was merged on September 4th (you need to hover the backport merge date- [Merge pull request](https://github.com/noobaa/noobaa-core/commit/7744b6f12d3ff3c9baef1fa2e9247b096af3bfe6) https://github.com/noobaa/noobaa-core/pull/8325 [from liranmauda/liran-backport-into-5.17](https://github.com/noobaa/noobaa-core/commit/7744b6f12d3ff3c9baef1fa2e9247b096af3bfe6))
